lib/DxilPIXPasses/DxilDbgValueToDbgDeclare.cpp

```diff
@@ -1207,24 +1231,28 @@ static llvm::Type *GetLLVMTypeFromDIBasicType(llvm::IRBuilder<> &B,
   return nullptr;
 }
 
-void VariableRegisters::PopulateAllocaMap_BasicType(llvm::DIBasicType *Ty) {
+void VariableRegisters::PopulateAllocaMap_BasicType(llvm::DIBasicType *Ty,
+                                                    unsigned sizeOverride) {
   llvm::Type *AllocaElementTy = GetLLVMTypeFromDIBasicType(m_B, Ty);
   assert(AllocaElementTy != nullptr);
   if (AllocaElementTy == nullptr) {
     return;
   }
 
-  const OffsetInBits AlignedOffset = m_Offsets.Add(Ty);
+  const auto offsets = m_Offsets.Add(Ty, sizeOverride);
 
   llvm::Type *AllocaTy = llvm::ArrayType::get(AllocaElementTy, 1);
-  llvm::AllocaInst *&Alloca = m_AlignedOffsetToAlloca[AlignedOffset];
-  Alloca = m_B.CreateAlloca(AllocaTy, m_B.getInt32(0));
-  Alloca->setDebugLoc(llvm::DebugLoc());
+  llvm::AllocaInst *&Alloca = m_AlignedOffsetToAlloca[offsets.Aligned];
+  if (Alloca == nullptr) {
+    Alloca = m_B.CreateAlloca(AllocaTy, m_B.getInt32(0));
+    Alloca->setDebugLoc(llvm::DebugLoc());
```

Comment:
Is this debug loc going to arbitrarily be assigned to the first member to get here? That is probably okay, but the first in the struct might be best and I don't know if that's guaranteed.

Reply:
I think this is OK. Multiple bitfields can live at the same aligned offset (that is, the offset of the actual storage). We'd have to choose some DebugLoc to associate the alloca with, so the first seems like as good a choice as any.